modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientKeyValueView.java

```diff
@@ -464,11 +464,12 @@ private HashMap<K, V> readGetAllResponse(ClientSchema schema, ClientMessageUnpac
 
         try {
             for (int i = 0; i < cnt; i++) {
-                in.unpackBoolean(); // TODO: Optimize (IGNITE-16022).
-
-                var tupleReader = new BinaryTupleReader(schema.columns().length, in.readBinaryUnsafe());
-                var reader = new ClientMarshallerReader(tupleReader);
-                res.put((K) keyMarsh.readObject(reader, null), (V) valMarsh.readObject(reader, null));
+                // TODO: Optimize (IGNITE-16022).
+                if (in.unpackBoolean()) {
```

Comment:
Does this boolean mean that the following value is not null?

Reply:
I think yes, this part was done by @ptupitsyn